big_tests/tests/mod_filter_SUITE.erl

```diff
@@ -0,0 +1,156 @@
+-module(mod_filter_SUITE).
+-compile(export_all).
+
+-include_lib("escalus/include/escalus.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+-import(distributed_helper, [mim/0,
+                             require_rpc_nodes/1,
+                             rpc/4]).
+%%--------------------------------------------------------------------
+%% Suite configuration
+%%--------------------------------------------------------------------
+
+all() ->
+    [{group, messages}].
+
+groups() ->
+    [{messages, [sequence], [
+                             allowed_messages_story,
+                             denied_messages_story
+                             ]}].
+
+suite() ->
+    escalus:suite().
+
+%%--------------------------------------------------------------------
+%% Init & teardown
+%%--------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    ok = dynamic_modules:ensure_modules(required_modules()),
+    update_rules(),
+    escalus:init_per_suite(Config).
+
+end_per_suite(Config) ->
+    escalus_fresh:clean(),
+    escalus:end_per_suite(Config).
+
+init_per_group(_GroupName, Config) ->
+    escalus:create_users(Config, escalus:get_users([alice, bob, astrid, kate])).
+
+end_per_group(_GroupName, Config) ->
+    escalus:delete_users(Config, escalus:get_users([alice, bob, astrid, kate])).
+
+init_per_testcase(CaseName, Config0) ->
+    NewUsers = proplists:get_value(escalus_users, Config0) ++ escalus_ct:get_config(escalus_anon_users),
+    Config = [{escalus_users, NewUsers}] ++ Config0,
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(CaseName, Config) ->
+    AnonJID = erlang:get(anon_user),
+    mongoose_helper:clear_last_activity(Config, AnonJID),
+    escalus:end_per_testcase(CaseName, Config).
+
+required_modules() ->
+  [{mod_filter, []}].
+
+% The configuration of rules is done using mim ACL and ACCESS.
+% mod_filter allows  limit traffic between domains or users based on their type.
+%
+% On this `update_rules` example, next rules are applied:
+%    1. the users of a private vhosts (localhost and sogndal)
+%       can only chat with themselves. so that particular host
+%       will have no connection to the exterior.
+%    2. anonymous users of some host (anonymous.localhost)
+%       can only chat with shared single chat user(some bot, i.e kate@localhost), 
+%       so that particular host will have no connection
+%       to the other hosts (localhost, sogndal) listed in hosts 
+update_rules()->
```

Comment:
This function is too long. If we've got a list of rules, let's put it into the list (returned from some funciton like `rules/0` or generally `update_rules_operations/0` and use list comprehension to match on `{atomic, ok}`.